dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -0,0 +1,200 @@
+    /// an agent specific version.
+    /// </summary>
+    /// <param name="agent">An optional agent, if requesting an agent history.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The message history</returns>
+    public IAsyncEnumerable<ChatMessageContent> GetChatMessagesAsync(Agent? agent = null, CancellationToken cancellationToken = default)
+    {
+        if (agent == null)
+        {
+            return this._history.ToDescendingAsync();
+        }
+
+        var channelKey = this.GetAgentHash(agent);
+        if (!this._agentChannels.TryGetValue(channelKey, out var channel))
+        {
+            return Array.Empty<ChatMessageContent>().ToAsyncEnumerable();
+        }
+
+        return channel.GetHistoryAsync(cancellationToken);
+    }
+
+    /// <summary>
+    /// Append messages to the conversation.
+    /// </summary>
+    /// <param name="message">Set of non-system messages with which to seed the conversation.</param>
+    /// <remarks>
+    /// Adding a message to the conversation requires any active <see cref="AgentChannel"/> remains
+    /// synchronized, so the message is broadcast to all channels.
+    /// </remarks>
+    /// <throws>KernelException if a system message is present, without taking any other action</throws>
+    public void AddChatMessage(ChatMessageContent message)
+    {
+        this.AddChatMessages(new[] { message });
+    }
+
+    /// <summary>
+    /// Append messages to the conversation.
+    /// </summary>
+    /// <param name="messages">Set of non-system messages with which to seed the conversation.</param>
+    /// <remarks>
+    /// Adding messages to the conversation requires any active <see cref="AgentChannel"/> remains
+    /// synchronized, so the messages are broadcast to all channels.
+    /// </remarks>
+    /// <throws>KernelException if a system message is present, without taking any other action</throws>
+    public void AddChatMessages(IReadOnlyList<ChatMessageContent> messages)
+    {
+        for (int index = 0; index < messages.Count; ++index)
+        {
+            if (messages[index].Role == AuthorRole.System)
+            {
+                throw new KernelException($"History does not support messages with Role of {AuthorRole.System}.");
+            }
+        }
+
+        // Append to chat history
+        this._history.AddRange(messages);
+
+        // Broadcast message to other channels (in parallel)
+        var channelRefs = this._agentChannels.Select(kvp => new ChannelReference(kvp.Value, kvp.Key));
+        this._broadcastQueue.Enqueue(channelRefs, messages);
+    }
+
+    /// <summary>
+    /// Add user message to chat history
+    /// </summary>
+    /// <param name="input">The user input, might be empty.</param>
+    public ChatMessageContent? AddUserMessage(string? input)
```

Comment:
Let's remove this, see the comments here regarding this pattern: https://github.com/microsoft/semantic-kernel/pull/5800#discussion_r1555816250